### PR TITLE
Added multi column sorting. Fully compatible with regular and backend…

### PR DESCRIPTION
**Pull request by suneab**
… sorting and pagination

Fixes #
#1694

## Proposed Changes

- Adds the posibility of sorting on multiple columns
- Compatible with both frontend and backend sorting
- Compatible with both frontend and backend pagination

### CODE REVIEW

**Comment by jtommy** [comment]
@suneab, please check my review.
I think we should add a way to reset the sort of a single column without resetting all.

**Comment by suneab** [comment]
@jtommy i just pushed a change to fix the .find IE11 issue
I also removed the lodash package and added my own custom sorting method. i added it to the helper class because .sort method was interfering with array.sort

How would you like the column sort resetting to work? 

the only idea i can come up with is
click column for sorting asc <- how it works now
click same coulmn for sorting desc <- how it works now
click same column to reset it <- new change (not implimented yet!)

but i don't like it.. 

would appreciate your input.

**Comment by suneab** [comment]
btw i left lodash orderBy for my fake backend sorting in the example. hope thats ok.

**Comment by jtommy** [comment]
@suneab I think it's a good example https://handsontable.com/docs/7.1.1/demo-multicolumn-sorting.html

**Comment by suneab** [comment]
I don't like it.. 
We keep the sorting priority when clicking the columns multiple time while only changing sorting direction, they change the priority every click.
If we added the reset on third click we would change the priority. 
Is there any other way of achieving this?

**Comment by suneab** [comment]
Hi @jtommy 
I would appreciate some feedback :)

**Comment by jtommy** [comment]
@suneab I don't know to be honest. Btw you should add a way to support multiple sort on mobile too (for example a dropdown)

**Comment by suneab** [comment]
@jtommy i just added mobile support (IE11 is supported as well). Please let me know whats missing for this feature to be approved.

**Comment by suneab** [comment]
oh btw, i couldnt figure out how to add the custom sorting icon to inside of the select options so i just added the HTML arrows
i found this thread, but its outdated and can't find anything in the current documentation about it
https://github.com/buefy/buefy/issues/74

**Comment by jtommy** [comment]
@suneab any news about reset sorting column ?

**Comment by suneab** [comment]
@jtommy nope, still dont know how to fit it in there

**Comment by jtommy** [comment]
We have to find a way to reset or a way to enable multi sort on each column

**Comment by suneab** [comment]
@jtommy i added a way to reset a single column by clicking a button. also works for mobile.

**Comment by jtommy** [comment]
@suneab Nice idea! I'm checking your code and `custom-key` prop is too generic name, please change it to `sort-multiple-key`

**Comment by suneab** [comment]
i fixed the naming of the custom key. Not sure what would be the correct versions in the package file

**Comment by jtommy** [comment]
@suneab i don't see your commit and please reset your package-lock.json

File: src/components/table/Table.vue [hunk 1]
```diff
@@ -91,7 +94,29 @@
                                 </template>
                                 <template v-else>{{ column.label }}</template>
 
+                                <template
+                                    v-if="sortMultiple &&
+                                        sortMultipleDataComputed.find(i =>
```
Comment (jtommy) [inline src/components/table/Table.vue]: Please don't use find method

File: src/components/table/Table.vue [hunk 2]
```diff
@@ -91,7 +94,29 @@
...
+                                    i.field === column.field)">
+                                    <b-icon
+                                        :icon="sortIcon"
+                                        :pack="iconPack"
+                                        both
+                                        :size="sortIconSize"
+                                        :class="{ 'is-desc': sortMultipleDataComputed.find(i =>
+                                        i.field === column.field).order === 'desc'}"
+                                    />
+                                    {{ sortMultipleDataComputed.findIndex(i =>
```
Comment (jtommy) [inline src/components/table/Table.vue]: Please don't use findIndex

File: src/components/table/Table.vue [hunk 3]
```diff
@@ -91,7 +94,29 @@
...
+                                    i.field === column.field) + 1 }}
+                                    <button
+                                        style="margin-left:10px"
```
Comment (jtommy) [inline src/components/table/Table.vue]: Please add a class that set margin and apply on it

**Comment by suneab** [comment]
> @suneab i don't see your commit and please reset your package-lock.json

lol i forgot to press commit.. i fixed the package as well

**Comment by jtommy** [comment]
@suneab please resolve conflicts so I'll merge it, thanks :)

**Comment by suneab** [comment]
conflicts resolved

**Comment by ms-saltation** [comment]
Thank you guys. Will try it out soon.

**Pull request merged**